src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -1103,7 +1103,7 @@ impl<'a, 'cfg> State<'a, 'cfg> {
                         // If this is an optional dependency, and the new feature resolver
                         // did not enable it, don't include it.
                         if dep.is_optional() {
-                            let features_for = unit_for.map_to_features_for(dep.artifact());
+                            let features_for = unit_for.map_to_features_for(None);
```

Comment:
Perhaps to stay consistent with the rest of the usage in this file, this const helps explain what the parameter is (instead of a nondescript `None`).

```suggestion
                            let features_for = unit_for.map_to_features_for(IS_NO_ARTIFACT_DEP);
```

Also, maybe a comment could be useful here to explain why `None` is being passed?